metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala

```diff
@@ -275,15 +276,41 @@ object MetalsEnrichments
         workspace.resolve(Directories.readonly).toNIO
       )
 
+    def toRelativeInside(prefix: AbsolutePath): Option[RelativePath] = {
+      val relative = path.toRelative(prefix)
+      if (relative.toNIO.getName(0).filename != "..") Some(relative)
+      else None
+    }
+
+    def jarPath: Option[AbsolutePath] = {
+      val filesystem = path.toNIO.getFileSystem()
+      if (filesystem.provider().getScheme().equals("jar")) {
```

Comment:
Do we also need to consider srcjars here? https://github.com/scalameta/metals/pull/2710

Reply:
Good catch. Thx!

Reply:
Ah ok, I wasn't 100% sure, that's why I brought it up.